src/controllers/zodiac.ts

```diff
@@ -0,0 +1,40 @@
+import { Request, Response } from "express";
+import { Signs_ZodiacDirBG } from "../models/signs";
+import {
+  GetLastElementJSNatFromSignData,
+  getTodayAllSignRecordsFromDB,
+} from "../dal/zodiacdeep";
+import { ISignRecord } from "../interfaces/signs";
+import { getText } from "../helpers/arrayProcessing";
+
+export async function getZodiac(req: Request, res: Response): Promise<void> {
+  {
+    const sign = req?.query?.sign;
```

Comment:
here you can cast sign to string, so you don't have to call it everywhere else. Example:

const sign = req?.query?.sign as string;